### ISSUE TITLE
Questions about mIoU and F-score

### ISSUE THREAD

**Issue by dreamtofree** [cyh-0/CAVP#6]
https://github.com/cyh-0/CAVP/blob/main/utils/eval_utils.py#L33
I observed that your calculations of evaluation metrics (mIoU, F-score) during testing are fundamentally different from previous works, such as AVSBench, AVSegformer and so on. For example, in your code, the miou is calculated as (i1+i2+...+in)/(u1+u2+...+un), and the previous method is (i1/u1+i2/u2+...+in/un)/n. Trough our experiments, your calculation methods of evaluation metrics improves the results significantly compared with previous methods. 
Above all, is it unfair in your paper to report SOTA using different calculations of evaluation metrics?

**Comment by cyh-0** [comment]
Hi @dreamtofree,

Thanks for rasing this issue. I accurately follow the traditional way in semantic segmenation to compute the mIoU to avoid

1) Using two different loss metrics for AVS-Object and AVS-Semantic.

2) Easy handling no-audio cases.

Additionally, this approach gives weight to each pixel equally, making it more representative of the overall segmentation performance across the entire dataset.

**Comment by dreamtofree** [comment]
Thank you for your reply. Maybe, there are some misunderstandings.
First of all, I have never heard that the traditional calculation method of IoU is (i1+i2+...+in)/(u1+u2+...+un). This is crazy. Please give me your reference. 
Secondly, in order to ensure a fair comparison, please use the metric calculation method of https://github.com/OpenNLPLab/AVSBench or https://github.com/vvvb-github/AVSegFormer instead of the wrong one. 
Your experimental results are unreasonable in my opinion, because I reproduced your S4 model. When using your metric, the IoU is 87.20 as reported in your paper, but when using the correct calculation method, it is even less than 70. My own model uses the correct calculation method to be 83+. But after changing to your calculation method, it rises to 91+.

**Comment by cyh-0** [comment]
Sorry about the confusion. I remember AVSBench mention they use Jaccard index (proposed by The Pascal VOC challenge). Hence follow the same way to compute the mIoU for all pixel in each class. 
Here are some examples from other segmentation repo:

https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/utils/metrics.py

https://github.com/VainF/DeepLabV3Plus-Pytorch/blob/master/metrics/stream_metrics.py

https://pytorch.org/ignite/_modules/ignite/metrics/confusion_matrix.html#mIoU

https://github.com/microsoft/ProDA/blob/9ba80c7dbbd23ba1a126e3f4003a72f27d121a1f/metrics.py#L25

https://github.com/open-mmlab/mmsegmentation/blob/main/mmseg/evaluation/metrics/iou_metric.py

The paper reports a result of 85.77 for the single-source split. I think you can highlight both results in your table for comparison.

Kind Regards

**Comment by Coding4AI-cyber** [comment]
True! @dreamtofree 

The comparison here is **indeed not fair**. Even though the author claimed the metrics used are from other conventional segmentation tasks, it is still an obligation to follow the original metrics used in AVSBench, which is the basic reason we call it a `benchmark`.

I have just transferred the metrics used here to the original method of AVS (AVSBench), and it reached a **remarkable increase** in the mIoU. However, this is not a fair comparison, as the benchmark should be evaluated using the original metrics defined for AVSBench.

Seriously, the authors should consider using the original metrics **without changing a single line of code**, for the best of our community.

**Comment by dreamtofree** [comment]
OK, fine. To be honest, I think you have understood what I am emphasing and why this is important.
I am not familiar with other repo, but mmsegmentation. It's true that it calculates iou the same with yours. However, as long as you set batch_size to 1 during testing, the calculation of iou in mmsegmentation is equivalent to (i1/u1+i2/u2+...+in/un)/n, which is also the common setting. In contrast, the calculation of iou in your code is (i1+i2+...+in)/(u1+u2+...+un) no matter what batch_size is set.
Going back to the fundamental issue, you used different calculation metrics to compare with previous work, which greatly broken the community of the AVSBench benchmark. (i1/u1+i2/u2+...+in/un)/n and (i1+i2+...+in)/(u1+u2+...+un) are just an abbreviation. In fact, the calculation method used in AVSBench takes into account several another aspects. Especially for AVSS subtask, the calculation method in AVSBench is very sensitive to category noise in predictions, and your calculation method completely circumvents this.
I noticed that you have published another ECCV2024 using the similar method and claimed to have obtained SOTA. My best wishes to you, please take https://github.com/ShiArthur03 as a warning, and keep our community fair and healthy.

**Comment by cyh-0** [comment]
Hi @dreamtofree,

Sorry for the misunderstanding about the metrics comment. I thought AVSBench was using the conventional segmentation metrics as in their paper they refer the Pascol VOC paper for the Jaccard Index and didn't specify they did certain modification for the evaluation metric. Hence I inheritant the traditional evaluation metrics from Pascol VOC and I didn't check the details in AVSBench's repo.

**Comment by Coding4AI-cyber** [comment]
@cyh-0 I think AVSBench-Semantic should also be revised, according to @dreamtofree .

> Especially for AVSS subtask, the calculation method in AVSBench is very sensitive to category noise in predictions, and your calculation method completely circumvents this. FROM  @dreamtofree 

I believe the issues extend beyond just the AVSBench-Object dataset. All of the results presented in Tables 1, 2, and 3 should be reviewed and revised, as the AVSBench-Semantic (AVSS) subtask also has its own unique sensitivities. 

As a member of the computer vision community, I strongly recommend that the author revise or even withdraw your papers published in CVPR and ECCV, since nearly all results need to be changed.

**Comment by dreamtofree** [comment]
Well, anyway, I accept your explanation. I know that it is unrealistic for you to withdraw your papers or revise the results in your papers. However, I hope you can keep this issue open to remind latecomers. In particular, if other papers are asked to compare with CAVP/CPM during peer-review, this issue can be used as cited material. At the same time, I hope you can correct your mistakes and do scientific research rigorously in the future.

**Comment by yyliu01** [comment]
Thanks for your interest.

@dreamtofree @Coding4AI-cyber 
According to Table 1, we believe there is a difference between the AVSBench and ordinary segmentation tasks. For example, the Cityscapes evaluation (https://www.cityscapes-dataset.com/benchmarks/) utilises the exact same (per-dataset calculation) method as ours. Since the author does not mention any implementation differences in their paper, we overlooked the evaluation code.  _We anticipate that the results based on their implementation will be updated in this repository and the arXiv version in the next few days._

Additionally, we believe there are issues with Table 1, as mentioned in our comments on their GitHub (https://github.com/OpenNLPLab/AVSBench/issues/13). Specifically, the images are mistakenly resized to 224x224, which leads to unnecessary non-rigid transformations ( in question 1). _We will also update this mistake with this setup in this repository and the arXiv version to avoid any confusion,  for the best of our community._

@Coding4AI-cyber 
Regarding Table 2 and Table 3, all results in these tables are based on the same evaluation protocol, in accordance with existing segmentation benchmarks via per-dataset mIoU calculation as mentioned above. We believe our implementation is correct and does not require any modifications. Please feel free to run the submitted code, and you are very welcome to post any issues.

Thank you for your time. We will let you know when the updates are made in this post.

**Comment by dreamtofree** [comment]
OK, I admit that in the segmentation community, there may indeed be situations where (i1+i2+...+in)/(u1+u2+...+un) is used to calculate IoU. However, if you check the AVSBench code carefully, this calculation method is much simpler than what is provided in the benchmark. 
I'm very much looking forward to your updated code and the results of the qualitative comparison, especially the AVSS subtask. I can give you a rough reference. I used your code for training and testing, and the mIoU accuracy was 50+, however when changing to the evaluation method in AVSBench it was only 40-.

**Comment by cyh-0** [comment]
Hi,

I have updated the results on AVS-Bench object dataset and results are as follows:
<p>
<del>
|                                        | AVSBench-Object (SS) |         | AVSBench-Object (MS) |         | AVSBench-Semantics |         |
|----------------------------------------|----------------------|---------|----------------------|---------|--------------------|---------|
|                                        | mIoU                 | F-Score | mIoU                 | F-Score | mIoU               | F-Score |
| TPAVI~\cite{zhou2022audio}             | 72.79                | 84.80   | 47.88                | 57.80   | 20.18              | 25.20   |
| AVSBG~\cite{hao2023improving}          | 74.13                | 85.40   | 44.95                | 56.80   | -                  | -       |
| ECMVAE~\cite{mao2023multimodal}        | 76.33                | 86.50   | 48.69                | 60.70   | -                  | -       |
| DiffusionAVS~\cite{mao2023contrastive} | 75.80                | 86.90   | 49.77                | 62.10   | -                  | -       |
| CAVP - CNN                                  | 76.94                | 87.31   | 52.68                | 64.87   | 30.37              | 35.29   |
</del>
</p>

<p>
<del>
avsbench-object-ss-224\
https://drive.google.com/file/d/1uBteMNAO_dXVgN8yzII901469jfJxcmO/view?usp=drive_link

avsbench-object-ms-224\
https://drive.google.com/file/d/1ZWE8dJV_uLfMrbnvZ8NW6GGwXM8-QhxA/view?usp=drive_link
</del>
</p>
avss-224\
https://drive.google.com/file/d/1DwVw_NtDv23QacpNvKlabWSnnPy25xfr/view?usp=drive_link

In terms of evaluation metrics, one important aspect to consider is that the evaluation batch size may affect the experimental results due to the max operation inside the f-score function. The current implementation of CAVP is based on a CNN architecture, and we will add transformer-based results later. I will update the modified version later.

Additionally, the current implementation of CAVP is simplified to use only AVSBench-Semantics. However, it is recommended to use the initial version of the AVSBench-Object dataset for training and evaluation, as AVSBench-Semantics has label noise issues in the dataset.

Lastly, for the AVS problem, training and evaluating the method on 224x224 images is not a desirable choice. As previously mentioned, **this can lead to unnecessary non-rigid transformations, loss of details, and distortion**. Using the original image resolution for training and evaluation is more commonly used and beneficial for real-world scenarios.

**Comment by Coding4AI-cyber** [comment]
1. I am not satisfied with your comparison, as there were still many excellent and high-performance works at the time that were not included in the comparison.
2. This problem exists in both your ECCV and CVPR papers, and not just in this one paper that needs to be updated.
3. The VPO-MSMI (stereo) dataset that was promised in your paper has not been released, or I just can't find it.
4. I will keep my eye on this project and your updated results.

**Comment by cyh-0** [comment]
I’ve updated the CAVP results in the README. I have updated the tables based on the AVSBench's metrics and the conventional semantic segmentation metrics. The table is further divided into **IMAGENET PRETRAINED** backbones and **COCO PRETRAINED** backbones for comparison. I will work on resolving the CPM issue and update its table as soon as possible.
The VPO datasets are available [`here`](https://drive.google.com/file/d/12jq7-Ke09ZPoUI1od44q97DNLrThoHc3/view?usp=sharing) and the VGGSound audio files are available [`here`](https://drive.google.com/file/d/1-OB3E9qbanfvZGbxvmRL05hsxwD0YOPq/view?usp=sharing). Position information will be added on the fly with the dataloader.

**Comment by Coding4AI-cyber** [comment]
The results of your updated experiments overlap with part of our reproduction, and I relatively acknowledge these results.

However, the results you have updated now are far from being the state-of-the-art when you submitted to CVPR last November, and are very behind advanced models like CATR  (published 2 months before CVPR ddl).

I believe you have improperly profited by following the wrong metrics, and have gotten papers published in CVPR and ECCV. I still hold a very opposing attitude towards your two papers. Although the author's attitude shows that this was an unintentional mistake, this improper profiting cannot be ignored by the community.

In summary, as a member of the community, I finally provide my **overall comments** to the author (Mr. Yuanhong Chen, Mr. Yuyuan Liu, **Prof. Gustavo Carneiro**). I do **not** think the author's behavior constitutes **academic misconduct**, but it is undeniable for **improper profiting**, which has allowed the paper to reach a level it does not deserve.
